### PR TITLE
fix: file#move do not delete origin file if same as destination

### PR DESCRIPTION
**Pull request by AVaksman**
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Currently if user passes same `bucket` object, `file`object, or `fileName` string as the original file, the client will attempt to "copy" (essentially leaving the file in place) and subsequently delete the origin/destination file.

### CODE REVIEW

**Comment by jkwlui** [comment]
Sounds like what we really should do here is to error out when the user tries to copy/move to the same destination, unless I'm missing something here?

**Comment by AVaksman** [comment]
IMHO I am not sure passing same destination on copy/move deserves an error (would be a breaking change). WDYT?

**Comment by jkwlui** [comment]
Yeah I would try to avoid a breaking change here as well. The draft looks good, promote it to a PR?

File: test/file.ts [hunk 1]
```diff
@@ -3459,11 +3460,40 @@ describe('File', () => {
         });
       });
 
+      it('should not delete the destination is same as origin', done => {
+        // sinon.stub(file, 'request').callsFake((callback: Function) => {
```
Comment (jkwlui) [inline test/file.ts]: Remove commented code.

Comment (AVaksman) [reply test/file.ts]: Thanks, done

**Pull request merged**